src/pages/receipts/csvFileReceipt.js

```diff
@@ -119,55 +119,58 @@ const csvFileButtonSubmit = () => {
             generateBSIqueryCSVData(modifiedResults);
             hideAnimation();
         } catch (e) {
             hideAnimation();
-            showNotifications({ title: 'Error', body: `Error fetching BSI Query Data -- ${e.message}` }, true);
+            showNotifications({ title: 'Error', body: `Error fetching BSI Query Data -- ${e.message}` });
         }
     });
 }
 
-const getCurrentDate = () => {
-  const currentDate = new Date().toLocaleDateString('en-CA');
-  return currentDate;
-}
-
-const getBSIQueryData = async (filter) => {
+const getSpecimensByReceivedDate = async (dateFilter) => {
     try {
         const idToken = await getIdToken();
-        const response = await fetch(`${baseAPI}api=queryBsiData&type=${filter}`, {
+        const response = await fetch(`${baseAPI}api=getSpecimensByReceivedDate&receivedTimestamp=${dateFilter}`, {
             method: "GET",
             headers: {
                 Authorization: "Bearer " + idToken,
             },
         });
 
         if (response.status !== 200) {
-            throw new Error(`Error fetching BSI Query Data. ${response.status}`);
+          throw new Error(`Error fetching specimens by received date. ${response.status}`);
         }
 
-        return await response.json();    
+        return await response.json();
     } catch (e) {
         console.error(e);
-        throw new Error(`Error fetching BSI Query Data: ${e.message}`);
+        throw new Error(`Error fetching specimens by received date: ${e.message}`);
     }
-};
+}
 
+const getCurrentDate = () => {
+  const currentDate = new Date().toLocaleDateString('en-CA');
+  return currentDate;
+}
 
 const modifyBSIQueryResults = (results) => {
-  let filteredResults = results.filter(result =>  result.length !== 0 && (result[0][fieldToConceptIdMapping.collectionId] !== undefined && 
-                        result[0][fieldToConceptIdMapping.collectionId].split(' ')[1] !== '0008' && result[0][fieldToConceptIdMapping.collectionId].split(' ')[1] !== '0009')
-                        && result[0][fieldToConceptIdMapping.discardFlag] !== fieldToConceptIdMapping.yes && result[0][fieldToConceptIdMapping.deviationNotFound] !== fieldToConceptIdMapping.yes)
-  filteredResults = filteredResults.flat()
-  filteredResults.forEach(filteredResult => {
-      let vialMappings = getVialTypesMappings(filteredResult)
-      updateResultMappings(filteredResult, vialMappings)
-  })
-  return filteredResults
+    const csvDataArray = [];
+    results.forEach(result => {
+        const collectionType = result[fieldToConceptIdMapping.collectionType];
+        const healthcareProvider = result[fieldToConceptIdMapping.healthcareProvider];
+        const specimenKeysArray = Object.keys(result.specimens);
+            for (const specimenKey of specimenKeysArray) {
+                const [collectionId, tubeId] = result.specimens[specimenKey][fieldToConceptIdMapping.collectionId].split(' ') ?? ['', ''];
```

Comment:
Nice assignment destructuring and accounting for undefined or null values

Reply:
Thanks